components/tidb_query_expr/src/lib.rs

```diff
@@ -91,10 +92,35 @@ fn map_compare_in_string_sig(ret_field_type: &FieldType) -> Result<RpnFnMeta> {
     })
 }
 
-fn map_like_sig(ret_field_type: &FieldType) -> Result<RpnFnMeta> {
+fn map_like_sig(ret_field_type: &FieldType, children: &[Expr]) -> Result<RpnFnMeta> {
+    // Must use the charset from args, but not the collator from ret_field_type.
+    // Because the TiDB doesn't push down the collation information when the new
+    // collation framework is not enabled, and always use the binary collation.
+    // However, the `_` pattern considers not only the order of strings, but
+    // also the number of characters. Some characters more than 1 bytes cannot
+    // be matched by `_` if the new collation framework is not enabled.
+    let target_collation = children[0]
```

Comment:
I'm not sure whether the arguments have been validated in this function, and whether I could get the `children[0]` and `children[1]` safely without validating the `children.len() >= 2` :thinking:. Could anyone confirm for me?